src/components/NavPage.js

```diff
@@ -1,12 +1,102 @@
 import { Helmet } from 'react-helmet';
-import React from "react";
-import SignIn from "./SignIn";
+import React, {useEffect, useState} from "react";
 import NavBar from "./NavBar";
+import WeatherInfo from "./WeatherInfo";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import Grid from "@mui/material/Grid";
+import {Image} from "@mui/icons-material";
 
 function NavPage() {
+    const key = "NQ6gDQ8Tb5v7rkrFyUH7yUunW5R2VRWh";
+    const [cityKey, setCityKey] = React.useState("");
+    const [weather, setWeather] = React.useState("");
+    const [temperature, setTemperature] = React.useState("");
+    const [cityText, setCityText] = useState("");
+    const [cityChange, setCityChange] = useState("");
+    //const [flag, setFlag] = useState(false);
+
+    //getting weather details
+    const gettingWeather = () => {
```

Comment:
This methods should not be on the NavPage, please move it to a different file that will be in charge of doing all the logic of getting data from the API and then use those methods please